cibyl/sources/elasticsearch/api.py

```diff
@@ -355,6 +364,125 @@ def get_deployment(self, **kwargs):
+                        }
+                    ]
+                }
+            },
+            "sort": [{"timestamp.keyword": {"order": "desc"}}]
+        }
+
+        build_numbers = []
+        if 'build_number' in kwargs:
+            build_numbers = [status.upper()
+                             for status in
+                             kwargs.get('build_number').value]
+
+        results = []
+        hits = []
+        for job in jobs_found:
+            query_body['query']['bool']['must'][0] = {
+                "match": {
+                    "job_name.keyword": f"{job}"
+                }
+            }
+            for build_number in build_numbers:
+                query_body['query']['bool']['must'][1]['bool']['should'] = {
+                    "match": {
+                        "build_num": build_number
+                    }
+                }
+                results = self.__query_get_hits(
+                    query=query_body,
+                    index='logstash_jenkins'
+                )
+
+                # We need to process all the results because
+                # in the same build we can have different tests
+                # So we can't use '"size": 1' in the query
+                if results:
+                    for result in results:
+                        hits.append(result)
+
+        if not results:
+            return jobs_found
+
+        for hit in hits:
+            job_name = hit['_source']['job_name']
+            build_number = hit['_source']['build_num']
+            test_name = hit['_source']['test_name']
+            test_status = hit['_source']['test_status']
+            class_name = hit['_source'].get(
+                'test_class_name',
+                None
+            )
+            # Some build is not parsed good and contains
+            # More info than a time in the field
+            try:
+                test_duration = hit['_source'].get(
+                    'test_time',
+                    None
+                )
+                if test_duration:
+                    test_duration = float(test_duration)*1000
+            except ValueError:
+                LOG.debug("'test_time' field is not well parsed in "
+                          "elasticsearch for job: %s and build ID: %s",
+                          job_name,
+                          build_number
+                          )
+                continue
+
+            build_object = Build(
+                build_id=build_number
+            )
+
+            jobs_found[job_name].add_build(build_object)
+            jobs_found[job_name].builds[build_number].add_test(
+                Test(
+                    name=test_name,
+                    result=test_status,
+                    duration=test_duration,
+                    class_name=class_name
+                )
+            )
+
+        return AttributeDictValue(
+            "jobs",
+            attr_type=Job,
+            value=jobs_found
```

Comment:
'jobs_found' is already an 'AttributeDictValue'. This output will not be usable.